server/audit/views.py

```diff
@@ -1,69 +1,202 @@
+        return serializer_class(*args, **kwargs)
+
     def create(self, request, *args, **kwargs):
         serializer = self.get_serializer(data=request.data)
         serializer.is_valid(raise_exception=True)
         saved_audit = serializer.save()
-        if saved_audit:
-            data = {'success': 'success'}
         if not saved_audit:
             return Response({'error': 'failed'}, status=status.HTTP_400_BAD_REQUEST)
-        return Response(data, status=status.HTTP_201_CREATED)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
 
     def list(self, request):
-        queryset = self.filter_queryset(self.get_queryset()).filter(
-                init_audit_id=request.query_params.get('init_audit_id'))
+        customuser_id = request.query_params.get('customuser_id')
+        init_audit_id = request.query_params.get('init_audit_id')
+
+        if customuser_id:
+            self.queryset = self.queryset.filter(customuser_id=customuser_id)
+        if init_audit_id:
+            self.queryset = self.queryset.filter(init_audit_id=init_audit_id)
+
+        serializer = self.get_serializer(self.queryset, many=True)
+        return Response(serializer.data)
+
+class RecordViewSet(viewsets.ModelViewSet):
+    http_method_names = ['post', 'get', 'patch', 'delete']
+    queryset = Record.objects.all()
+    permission_classes = []
+
+    def get_permissions(self):
+        factory = PermissionFactory(self.request)
+        permission_classes = factory.get_general_permissions(
+            [HasSameOrgInQuery, ValidateSKOfSameOrg])
+        return [permission() for permission in permission_classes]
+
+    def get_serializer(self, *args, **kwargs):
+        serializer_class = RecordSerializer
+        return serializer_class(*args, **kwargs)
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        record = serializer.save()
+        if not record:
+            return Response({'error': 'failed'}, status=status.HTTP_400_BAD_REQUEST)
+        set_audit_accuracy(record.audit.audit_id)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def update(self, request, *args, **kwargs): # pylint: disable=unused-argument
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+        set_audit_accuracy(instance.audit.audit_id)
+        return Response(serializer.data)
+
+    def destroy(self, request, *args, **kwargs): # pylint: disable=unused-argument
+        instance = self.get_object()
+        self.perform_destroy(instance)
+        set_audit_accuracy(instance.audit.audit_id)
+        return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
why don't you call the super class' implementation instead of rewriting the code here?

Reply:
Also, I just looked into this, you could actually just override the dispatch method instead. Would be much cleaner and you could delete all these methods.

[See this SO post for details](https://stackoverflow.com/questions/59331300/invoke-function-after-all-methods-django-rest-framework-modelviewset )

Reply:
i want to be able to set the accuracy and maybe do other stuff before returning the response

Reply:
> 
> 
> i want to be able to set the accuracy and maybe do other stuff before returning the response

For anything before we'll have to have the methods defined, Nick's suggestion is simply a method that runs after all methods in the viewset, we'll have to get on call to figure out if this is what we can use optimally

Reply:
yeah, with the dispatch, it should help you do that without having to override every method. It's like a hook.